### PR TITLE
sensor: ti_hdc: convert driver to new devicetree APIs

### PR DESCRIPTION
**Pull request by galak**

### CODE REVIEW

File: drivers/sensor/ti_hdc/ti_hdc.h [hunk 1]
```diff
@@ -23,14 +23,14 @@
...
 struct ti_hdc_config {
 	struct i2c_dt_spec i2c;
+	struct gpio_dt_spec drdy;
 };
 
 struct ti_hdc_data {
 	uint16_t t_sample;
 	uint16_t rh_sample;
 
 #if DT_INST_NODE_HAS_PROP(0, drdy_gpios)
```
Comment (gmarull) [inline drivers/sensor/ti_hdc/ti_hdc.h:33]: this is now uncoditional, it wasn't before, also, `drdy` is not initialized

File: drivers/sensor/ti_hdc/ti_hdc.c [hunk 2]
```diff
@@ -23,14 +23,13 @@ LOG_MODULE_REGISTER(TI_HDC, CONFIG_SENSOR_LOG_LEVEL);
 static void ti_hdc_gpio_callback(const struct device *dev,
 				  struct gpio_callback *cb, uint32_t pins)
 {
+	const struct ti_hdc_config *cfg = dev->config;
```
Comment (gmarull) [inline drivers/sensor/ti_hdc/ti_hdc.c:26]: `dev` is a GPIO device here

**Review by gmarull — CHANGES_REQUESTED** [review]

File: drivers/sensor/ti_hdc/ti_hdc.c (hunk 2 above)
Comment (galak) [reply drivers/sensor/ti_hdc/ti_hdc.c]: fixed

File: drivers/sensor/ti_hdc/ti_hdc.h (hunk 1 above)
Comment (galak) [reply drivers/sensor/ti_hdc/ti_hdc.h]: Fixed.

**Review by MaureenHelm — APPROVED** [review]

**Review by gmarull — APPROVED** [review]

**Pull request merged**